src/managers/config.py

```diff
@@ -101,97 +94,20 @@ def advertised_listener(self) -> str:
-        )
-        self.jmx_prometheus_config_filepath = f"{self.charm.snap.CONF_PATH}/jmx_prometheus.yaml"
-
-    @property
-    def internal_user_credentials(self) -> Dict[str, str]:
-        """The charm internal usernames and passwords, e.g `sync` and `admin`.
-
-        Returns:
-            Dict of usernames and passwords
-        """
-        credentials = {
-            user: password
-            for user in INTERNAL_USERS
-            if (password := self.charm.get_secret(scope="app", key=f"{user}-password"))
-        }
-
-        if not len(credentials) == len(INTERNAL_USERS):
-            return {}
-
-        return credentials
-
-    @property
-    def zookeeper_config(self) -> Dict[str, str]:
-        """The config from current ZooKeeper relations for data necessary for broker connection.
-
-        Returns:
-            Dict of ZooKeeeper:
-            `username`, `password`, `endpoints`, `chroot`, `connect`, `uris` and `tls`
-        """
-        zookeeper_config = {}
-        # loop through all relations to ZK, attempt to find all needed config
-        for relation in self.charm.model.relations[ZK]:
-            if not relation.app:
-                continue
-
-            zk_keys = ["username", "password", "endpoints", "chroot", "uris", "tls"]
-            missing_config = any(
-                relation.data[relation.app].get(key, None) is None for key in zk_keys
-            )
-
-            # skip if config is missing
-            if missing_config:
-                continue
-
-            # set if exists
-            zookeeper_config.update(relation.data[relation.app])
-            break
-
-        if zookeeper_config:
-            sorted_uris = sorted(
-                zookeeper_config["uris"].replace(zookeeper_config["chroot"], "").split(",")
-            )
-            sorted_uris[-1] = sorted_uris[-1] + zookeeper_config["chroot"]
-            zookeeper_config["connect"] = ",".join(sorted_uris)
-
-        return zookeeper_config
-
-    @property
-    def zookeeper_related(self) -> bool:
-        """Checks if there is a relation with ZooKeeper.
-
-        Returns:
-            True if there is a ZooKeeper relation. Otherwise False
-        """
-        return bool(self.charm.model.relations[ZK])
-
-    @property
-    def zookeeper_connected(self) -> bool:
-        """Checks if there is an active ZooKeeper relation with all necessary data.
-
-        Returns:
-            True if ZooKeeper is currently related with sufficient relation data
-                for a broker to connect with. Otherwise False
-        """
-        if self.zookeeper_config.get("connect", None):
-            return True
-
-        return False
+    def __init__(
+        self,
+        state: ClusterState,
+        workload: KafkaWorkload,
+        config: CharmConfig,
+        current_version: str,
+    ):
+        self.state = state
+        self.workload = workload
+        self.config = config
+        self.current_version = current_version
```

Comment:
**todo:** I'm not sure this needs to be part of the constructor, as it's a part of the state, right? Couldn't we do:

```python
self.state.zookeeper.get_zookeeper_version()
```

Reply:
Current version comes from `events.upgrade` since it is parsed through the `DependencyModel`

Reply:
Right, could we add it to the Zookeeper state though?

Reply:
Sorry, this is Kafka state, not zookeeper one.